### PR TITLE
executor: fix the bug of step value overflow when split regions

### LINKED ISSUE #23159: Step value overflowed on split command
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
mysql> create table if not exists `t` (`id` bigint);
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> split table `t` between (-9223372036854775808) and (9223372036854775807) regions 4;
ERROR 1105 (HY000): Split table `t` region step value should more than 1000, step 0 is invalid
```
This happens because `step` value is overflowed while calculating it.
https://github.com/pingcap/tidb/blob/v4.0.8/executor/split.go#L542

### 2. What did you expect to see? (Required)

The command succeeded, and the table is splitted. (really?)

### 3. What did you see instead (Required)

```
ERROR 1105 (HY000): Split table `t` region step value should more than 1000, step 0 is invalid
```

### 4. What is your TiDB version? (Required)

v4.0.8

### PR DESCRIPTION
**Pull request by mantuliu**
Signed-off-by: mantuliu <240951888@qq.com>

### What problem does this PR solve?

Issue Number: close #23159 

Problem Summary:

### What is changed and how it works?

How it Works:
Use math/big package.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`: no
- Need to cherry-pick to the release branch: yes

### Check List 

Tests 

- Unit test
- Integration test 

### Release note 
- Fix the bug of step value overflow when split regions.

### CODE REVIEW

**Comment by mantuliu** [comment]
@wshwsh12 PTAL thanks.

File: executor/split_test.go [hunk 1]
```diff
@@ -366,6 +366,36 @@ func (s *testSplitIndex) TestSplitTable(c *C) {
...
+		},
+	}
+	ctx := mock.NewContext()
+	e := &SplitTableRegionExec{
+		baseExecutor: newBaseExecutor(ctx, nil, 0),
+		tableInfo:    tbInfo,
+		handleCols:   core.NewIntHandleCols(&expression.Column{RetType: types.NewFieldType(mysql.TypeLonglong)}),
+		lower:        []types.Datum{types.NewDatum(-9223372036854775808)},
+		upper:        []types.Datum{types.NewDatum(9223372036854775807)},
+		num:          4,
```
Comment (occho) [inline executor/split_test.go:391]: What happens when `REGIONS 1` is given?

Comment (mantuliu) [reply executor/split_test.go]: I add some edge test. And in the Integration Testing, the mysql client get the result:
```
mysql> create table if not exists `t1` (`id` bigint);
Query OK, 0 rows affected (0.00 sec)

mysql> split table `t1` between (-9223372036854775808) and (9223372036854775807) regions 1;
ERROR 1105 (HY000): Split table `t1` region step value should more than 1000, step -1 is invalid
```

**Comment by AilinKid** [comment]
why close it? @mantuliu

**Comment by mantuliu** [comment]
@AilinKid https://github.com/pingcap/tidb/pull/21277 fix it

**Pull request closed without merging**